### PR TITLE
DHFPROD-6528: Unable to find Source Records error when updating XPATH

### PR DESCRIPTION
**Pull request by briantang**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by Sanjeevani19 — COMMENTED** [review]
Works as expected, waiting for the tests to pass

**Review by xnikhil08 — APPROVED** [review]
Looks good. Unit tests pass locally. Hope the e2e tests pass as well, in the pipeline.

**Review by ngodugu-marklogic — DISMISSED** [review]
LGMT!

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.tsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ const MappingCard: React.FC<Props> = (props) => {
   useEffect(() => {
     let sortedArray = props.data.length > 1 ? sortStepsByUpdated(props.data) : props.data;
     setSortedMappings(sortedArray);
-    setSourceData([]);
+    // setSourceData([]);
```
Comment (rjrudin) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.tsx:93]: If this line of code isn't needed - it apparently is causing a problem - let's delete it! Commented-out code is generally an anti-pattern; we have branches and version history for storing code we're not ready to integrate yet.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.tsx (hunk 1 above)
Comment (briantang) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-card.tsx]: Thanks @rjrudin, removed the faulty line vs commenting it.

**Review by rjrudin — APPROVED** [review]

**Review by Sanjeevani19 — DISMISSED** [review]

**Review by srinathgit — CHANGES_REQUESTED** [review]
Couple of e2e test failures
https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-5124/6/tests

**Review by Sanjeevani19 — DISMISSED** [review]

**Review by srinathgit — CHANGES_REQUESTED** [review]
2 e2e failures
https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-5124/8/tests

**Comment by briantang** [comment]
Due to step cards sorting within the component on a useEffect, cypress failed when looking for certain locators mid-sort.

Refactored step-sorting code so that step artifacts are sorted (by last updated) before being passed into each component, much more efficient and less buggy. Cypress tests are passing now.

**Review by srinathgit — APPROVED** [review]

**Pull request merged**